ansible/roles/debops.ansible_plugins/filter_plugins/debops_filter_plugins.py

```diff
@@ -430,3 +422,434 @@ def filters(self):
             'parse_kv_config': parse_kv_config,
             'parse_kv_items': parse_kv_items
         }
+
+
+if __name__ == '__main__':
+    import unittest
+    import textwrap
+    import yaml
+
+    class Test(unittest.TestCase):
+
+        def test_parse_kv_value_simple(self):
+            current_data = yaml.safe_load(textwrap.dedent('''
+            name: 'local'
+            value: 'test'
+            '''))
+
+            new_data = yaml.safe_load(textwrap.dedent('''
+            name: 'local'
+            value: 'test2'
+            '''))
+
+            _parse_kv_value(current_data, new_data, 0)
+
+            #  print(yaml.dump(current_data, default_flow_style=False))
+            #  print(yaml.dump(expected_items, default_flow_style=False))
+
+            self.assertEqual(current_data, new_data)
+
+        def test_parse_kv_value_mixed(self):
+            current_data = yaml.safe_load(textwrap.dedent('''
+            name: 'local'
+            value:
+              - 'alpha'
+              - 'test'
+            '''))
+
+            new_data = yaml.safe_load(textwrap.dedent('''
+            name: 'local'
+            value:
+              - 'beta'
+            '''))
+
+            # FIXME: Not getting it. Why does the _parse_kv_value behave like
+            # this?
+            expected_data = yaml.safe_load(textwrap.dedent('''
+            name: local
+            value:
+              beta:
+                id: 0
+                name: beta
+                real_weight: 0
+                state: present
+                weight: 0
+            '''))
```

Comment:
I think that this might be a bug. From the top of my head I don't remember any roles that depend on this data structure, but I would think that this should be a list, and not a dictionary, since the `value`'s value is a YAML list. Then the "value" can be extracted from each dictionary `name` key, which is done in a bunch of roles already.